@xen-orchestra/web/src/components/pool/connect/ConnectionForm.vue

```diff
@@ -87,34 +95,45 @@ const form = reactive<NewServerForm>({
   allowUnauthorized: false,
 })
 
-async function submit() {
-  connecting.value = true
-
-  const optionalFields = Object.assign(
+const payload = computed(() => ({
+  host: form.host,
+  username: form.username,
+  password: form.password,
+  ...Object.assign(
     {},
     form.httpProxy && { httpProxy: form.httpProxy },
     form.readOnly && { readOnly: form.readOnly },
     form.allowUnauthorized && { allowUnauthorized: form.allowUnauthorized }
-  )
-
-  const payload = {
-    host: form.host,
-    username: form.username,
-    password: form.password,
-    ...optionalFields,
-  }
-
+  ),
+}))
+
+// TODO: multiple server creation not possible in the UI for now
+// so only handle a single payload
+const { canRun: createCanRun, isRunning: createIsRunning, run: create } = useJobServerCreate([payload])
+const { isRunning: connectIsRunning, run: connect } = useJobServerConnect([serverId])
+const { isRunning: removeIsRunning, run: remove } = useJobServerRemove([serverId])
```

Comment:
```suggestion
const { isRunning: removeIsRunning, run: remove } = useJobServerRemove([serverId])

const isSubmitting = computed(() => createIsRunning.value || connectIsRunning.value || removeIsRunning.value)

```

maybe we can simplify like this ?

Reply:
Or:
```ts
const isServerJobRunning = logicOr(createIsRunning || connectIsRunning || removeIsRunning)
```

Reply:
Indeed ^